cmd/earthly/app/run.go

```diff
@@ -304,18 +303,16 @@ func (app *EarthlyApp) run(ctx context.Context, args []string) int {
 			}
 			return 2
 		case isInterpreterError:
-			app.BaseCLI.Logbus().Run().SetFatalError(time.Now(), ie.TargetID, "", logstream.FailureType_FAILURE_TYPE_SYNTAX, ie.Error())
-			app.BaseCLI.Console().VerboseWarnf("Error: %s\n", ie.Error())
-			return 1
-		default:
-			if app.BaseCLI.CommandName() == "build" {
-				app.BaseCLI.Console().VerboseWarnf("Error: %v\n", err)
-				app.BaseCLI.Logbus().Run().SetFatalError(time.Now(), "", "", logstream.FailureType_FAILURE_TYPE_OTHER, err.Error())
+			app.BaseCLI.Console().Printf("Identified interperter error")
+			if ie.TargetID == "" {
+				app.BaseCLI.Logbus().Run().SetGenericFatalError(time.Now(), logstream.FailureType_FAILURE_TYPE_SYNTAX, errorWithPrefix(ie.Error()))
 			} else {
-				app.BaseCLI.Logbus().Run().SkipFatalError()
```

Comment:
Hey @mikejholly ,
i paired with @vladaionescu on this part and we decided to remove `SkipFatalError()` and use `SetFatalError()`.
I was wondering if you can share why the former was needed in the first place. I'd like to make sure I didn't miss anything with this change.

Reply:
The move to Logstream led to some errors being printed twice. I wanted to remove one, but that led to `No SetFatalError called appropriately` being called via the `defer`. Hence, the new "skip" method.

Example: 
```
            _unknown *failed* | build target: build main: failed to solve: Earthfile line 0:0 target asdfasdfasdf not found
            _unknown *failed* | in		github.com/earthly/earthly:mh/enable-logstream-no-upload+asdfasdfasdf
Error: build target: build main: failed to solve: Earthfile line 0:0 target asdfasdfasdf not found
in		github.com/earthly/earthly:mh/enable-logstream-no-upload+asdfasdfasdf
```

Reply:
ok, so sounds like this change should be ok?
I instead removed the call to `app.BaseCLI.Console().Warnf("Error: %v\n", err)`.

BTW, this error would now look like this:
```
View logs at https://cloud.earthly.dev/builds/28ee7b18-8116-4362-88d1-7d9bcf2ad34e
Error: /var/folders/v7/4fy0kdvj4hd_kvgqctg0_qwc0000gn/T/earthly-git870944423/Earthfile line 0:0 target asdfasdfasdf not found
in              github.com/earthly/earthly+asdfasdfasdf

```

Reply:
LGTM :+1: